packages/react-native-session-replay/ios/Tests/RCTTextViewRecorderTests.swift

```diff
@@ -0,0 +1,188 @@
+
+        let element = try XCTUnwrap(result as? SessionReplaySpecificElement)
+        XCTAssertEqual(element.subtreeStrategy, .ignore)
+        XCTAssertEqual(element.nodes.count, 1)
+        let wireframe = try XCTUnwrap(element.nodes[0].wireframesBuilder.buildWireframes(with: .init())[0].getAsTextWireframe())
+        XCTAssertEqual(wireframe.text, "This is the nested test text.")
+    }
+    
+    func testReturnsBuilderWithCorrectInformationWhenTextIsObfuscated() throws {
+        let mockMaskContext = SessionReplayViewTreeRecordingContext(
+            recorder: .init(privacy: SessionReplayPrivacyLevel.mask, applicationID: "app_id", sessionID: "session_id", viewID: "view_id", viewServerTimeOffset: nil),
+            coordinateSpace: UIView(),
+            ids: .init(),
+            imageDataProvider: ImageDataProvider()
+        )
+        let reactTag = NSNumber(value: 44)
+        let uiManagerMock = MockUIManager(reactTag: reactTag, shadowView: mockShadowView)
+        let viewMock = RCTTextView()
+        viewMock.reactTag = reactTag
+        let viewRecorder = RCTTextViewRecorder(uiManager: uiManagerMock)
+
+        let result = viewRecorder.semantics(of: viewMock, with: mockAttributes, in: mockMaskContext)
+
+        let element = try XCTUnwrap(result as? SessionReplaySpecificElement)
+        XCTAssertEqual(element.subtreeStrategy, .ignore)
+        XCTAssertEqual(element.nodes.count, 1)
+        let wireframe = try XCTUnwrap(element.nodes[0].wireframesBuilder.buildWireframes(with: .init())[0].getAsTextWireframe())
+        XCTAssertEqual(wireframe.text, "xxxx xx xxx xxxx xxxxx")
+    }
+}
+
+private class MockRCTTextView: RCTTextView {}
+
+private class MockUIManager: RCTUIManager {
+    /// Tag to be used in the test corresponding to a shadow view
+    var shadowViewTag: NSNumber? = nil
+    var shadowView: RCTTextShadowView? = nil
+    
+    convenience init(reactTag: NSNumber, shadowView: RCTTextShadowView?) {
+        self.init()
+        self.shadowViewTag = reactTag
+        self.shadowView = shadowView
+    }
+
+    internal override func shadowView(forReactTag: NSNumber) -> RCTShadowView? {
+        if (forReactTag == shadowViewTag) {
+            return shadowView
+        }
+        return nil
+    }
+    
+}
+
+extension SessionReplayInvisibleElement: Equatable {
+    public static func ==(lhs: SessionReplayInvisibleElement, rhs: SessionReplayInvisibleElement) -> Bool {
+        // If two elements are indeed InvisibleElement they're InvisibleElement.constant
+        return true
+    }
+}
+
+extension SRWireframe {
+    public func getAsTextWireframe() -> SRTextWireframe? {
+        switch self {
+        case .shapeWireframe(_):
+            return nil
+        case .textWireframe(let value):
+            return value
+        case .imageWireframe(_):
+            return nil
+        case .placeholderWireframe(_):
+            return nil
+        }
+    }
+}
```

Comment:
this feels a bit hacky, if you see a better way to do that I'm happy to know about it

Reply:
You can shorten it this way, but unfortunately, that's how deal with associated values in enums:

```swift
extension SRWireframe {
    public func getAsTextWireframe() -> SRTextWireframe? {
        if case .textWireframe(let value) = self {
            return value
        }
        return nil
    }
}
```